posts/2023-10-05-Rust-1.73.0.md

```diff
@@ -0,0 +1,114 @@
+
+If you have a previous version of Rust installed via rustup, you can get 1.73.0 with:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.73.0](https://github.com/rust-lang/rust/releases/tag/1.73.0) on GitHub.
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.73.0 stable
+
+## Cleaner panic messages
+
+The output produced by the default panic handler has been changed
+to put the panic message on its own line instead of wrapping it in quotes.
+This can make panic messages easier to read, as shown in this example:
+
+<div style="margin:1em"><pre><code>fn main() {
+    let file = "ferris.txt";
+    panic!("oh no! {file:?} not found!");
+}</code></pre>
+Output before Rust 1.73:
+<pre style="margin-top:0"><code style="background:#000;color:#ccc" class="language-text">thread 'main' panicked at 'oh no! "ferris.txt" not found!', src/main.rs:3:5</code></pre>
+Output starting in Rust 1.73:
+<pre style="margin-top:0"><code style="background:#000;color:#ccc" class="language-text">thread 'main' panicked at src/main.rs:3:5:
+oh no! "ferris.txt" not found!</code></pre></div>
+
+This is especially useful when the message is long, contains nested quotes, or spans multiple lines.
+
+Additionally, the panic messages produced by `assert_eq` and `assert_ne` have
+been modified, moving the custom message (the third argument)
+and removing some unnecessary punctuation, as shown below:
+
+<div style="margin:1em"><pre><code>fn main() {
+    assert_eq!("🦀", "🐟", "ferris is not a fish");
+}</code></pre>
+Output before Rust 1.73:
+<pre style="margin-top:0"><code style="background:#000;color:#ccc" class="language-text">thread 'main' panicked at 'assertion failed: `(left == right)`
+ left: `"🦀"`,
+right: `"🐟"`: ferris is not a fish', src/main.rs:2:5</code></pre>
+Output starting in Rust 1.73:
+<pre style="margin-top:0"><code style="background:#000;color:#ccc" class="language-text">thread 'main' panicked at src/main.rs:2:5:
+assertion `left == right` failed: ferris is not a fish
+ left: "🦀"
+right: "🐟"</code></pre></div>
+
+### Thread local initialization
+
+As proposed in [RFC 3184](https://github.com/rust-lang/rfcs/blob/master/text/3184-thread-local-cell-methods.md), `LocalKey<Cell<T>>` and `LocalKey<RefCell<T>>` can now be directly manipulated with `get()`, `set()`, `take()`, and `replace()` methods without jumping through a `with(|inner| ...)` closure which is necessary for general `LocalKey` work. `LocalKey<T>` is the type of `thread_local!` statics.
```

Comment:
This is a complicated sentence... maybe this reads better?
```suggestion
As proposed in [RFC 3184](https://github.com/rust-lang/rfcs/blob/master/text/3184-thread-local-cell-methods.md), `LocalKey<Cell<T>>` and `LocalKey<RefCell<T>>` can now be directly manipulated with `get()`, `set()`, `take()`, and `replace()` methods, rather than jumping through a `with(|inner| ...)` closure as needed for for general `LocalKey` work. `LocalKey<T>` is the type of `thread_local!` statics.
```

Or perhaps it should be split into multiple sentences.

Reply:
Yeah, not a fan of it either. I committed your suggestion, I probably won't have time to refactor into multiple sentences. I didn't have much luck making this look nice when I wrote it initially.